mathics/core/symbols.py

```diff
@@ -0,0 +1,783 @@
+                            "System`Postfix",
+                            Expression("System`List", leaves[0]),
+                            "..",
+                            170,
+                        ),
+                    )
+                else:
+                    return Expression("System`HoldForm", expr)
+            elif head == "System`RepeatedNull":
+                if len(leaves) == 1:
+                    return Expression(
+                        "System`HoldForm",
+                        Expression(
+                            "System`Postfix",
+                            Expression("System`List", leaves[0]),
+                            "...",
+                            170,
+                        ),
+                    )
+                else:
+                    return Expression("System`HoldForm", expr)
+
+            # If expr is not an atom, looks for formats in its definition
+            # and apply them.
+            def format_expr(expr):
+                if not (expr.is_atom()) and not (expr.head.is_atom()):
+                    # expr is of the form f[...][...]
+                    return None
+                name = expr.get_lookup_name()
+                formats = evaluation.definitions.get_formats(name, form)
+                for rule in formats:
+                    result = rule.apply(expr, evaluation)
+                    if result is not None and result != expr:
+                        return result.evaluate(evaluation)
+                return None
+
+            formatted = format_expr(expr)
+            if formatted is not None:
+                result = formatted.do_format(evaluation, form)
+                if include_form:
+                    result = Expression(form, result)
+                result.unformatted = unformatted
+                return result
+
+            # If the expression is still enclosed by a Format,
+            # iterate.
+            # If the expression is not atomic or of certain
+            # specific cases, iterate over the leaves.
+            head = expr.get_head_name()
+            if head in formats:
+                expr = expr.do_format(evaluation, form)
+            elif (
+                head != "System`NumberForm"
+                and not expr.is_atom()
+                and head != "System`Graphics"
+                and head != "System`Graphics3D"
+            ):
+                # print("Not inside graphics or numberform, and not is atom")
+                new_leaves = [leaf.do_format(evaluation, form) for leaf in expr.leaves]
+                expr = Expression(expr.head.do_format(evaluation, form), *new_leaves)
+
+            if include_form:
+                expr = Expression(form, expr)
+            expr.unformatted = unformatted
+            return expr
+        finally:
+            evaluation.dec_recursion_depth()
+
+    def format(
+        self, evaluation, form, **kwargs
+    ) -> typing.Union["Expression", "Symbol"]:
```

Comment:
I think `BaseExpression` is used, we don't need the Union or strings. And flake8 seems to be hapier.